src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotDisableCertificateValidationTests.cs

```diff
@@ -0,0 +1,448 @@
+        Return AcceptAllCertifications2(sender, certification, chain, sslPolicyErrors)
+    End Function
+
+    Function AcceptAllCertifications2(ByVal sender As Object, ByVal certification As System.Security.Cryptography.X509Certificates.X509Certificate, ByVal chain As System.Security.Cryptography.X509Certificates.X509Chain, ByVal sslPolicyErrors As System.Net.Security.SslPolicyErrors) As Boolean
+        Return True
+    End Function
+End Module");
+        }
+
+        [Fact]
+        public void TestDelegateCreationFromLocalFromLocal2NoDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Net;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+class TestClass
+{
+    public bool AcceptAllCertifications2(
+          object sender,
+          X509Certificate certificate,
+          X509Chain chain,
+          SslPolicyErrors sslPolicyErrors)
+    {
+        return true;
+    }
+
+    public bool AcceptAllCertifications(
+          object sender,
+          X509Certificate certificate,
+          X509Chain chain,
+          SslPolicyErrors sslPolicyErrors)
+    {
+        var a = 5;
+        if(a > 1)
+        {
+            return true;
+        }
+        else
+        {
+            return AcceptAllCertifications2(
+              sender,
+              certificate,
+              chain,
+              sslPolicyErrors);
+        }
+    }
+
+    public void TestMethod()
+    {
+        ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback(AcceptAllCertifications);
+    }
+}");
+        }
+
+        private void VerifyAcrossTwoAssemblies(string source1, string source2, string language, params DiagnosticResult[] expected)
+        {
+            Debug.Assert(language == LanguageNames.CSharp || language == LanguageNames.VisualBasic);
+
+            Project project1 = CreateProject(new[] { source1 }, language: language, referenceFlags: ReferenceFlags.RemoveCodeAnalysis);
+            Project project2 = CreateProject(new[] { source2 }, language: language, referenceFlags: ReferenceFlags.RemoveCodeAnalysis, addToSolution: project1.Solution)
+                           .AddProjectReference(new ProjectReference(project1.Id));
+
+            DiagnosticAnalyzer analyzer = language == LanguageNames.CSharp ? GetCSharpDiagnosticAnalyzer() : GetBasicDiagnosticAnalyzer();
+            GetSortedDiagnostics(analyzer, project2.Documents.ToArray()).Verify(analyzer, GetDefaultPath(language), expected);
+        }
+
+        private void VerifyCSharpAcrossTwoAssemblies(string source1, string source2, params DiagnosticResult[] expected)
+        {
+            VerifyAcrossTwoAssemblies(source1, source2, LanguageNames.CSharp, expected);
```

Comment:
This actually adds both the source files to the same assembly. You probably want to extend the test library that @dotpaul added recently for your scenario. Paul, can you help @LingxiaChen here?

Reply:
Ah, I notice you create two projects in the helper above - that works fine for me!

Reply:
Dangit, I think I like @LingxiaChen's approach better.  :-)